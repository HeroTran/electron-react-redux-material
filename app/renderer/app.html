<!DOCTYPE html>
<html lang="en">
  <head>
    <title>herotran210_desktop_app</title>
    <meta name="description" content="herotran210_desktop_app">
    <meta name="keywords" content="herotran210_desktop_app">
    <meta name="author" content="herotran210_desktop_app">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <link rel="icon" type="image/png" sizes="32x32" href="../../resources/favicon.ico" />
    <link rel="icon" type="image/png" sizes="16x16" href="../../resources/favicon.ico" />
    <link rel="shortcut icon"  href="../../resources/favicon.ico" />
    <script>
      (() => {
        if (
          typeof process !== 'object' ||
          (typeof process === 'object' && !process.env.START_HOT)
        ) {
          const link = document.createElement('link');
          link.rel = 'stylesheet';
          link.href = '../dist/style.css?v' + Math.random();
          document.getElementsByTagName('head')[0].appendChild(link);
        }
      })();
    </script>
  </head>
  <body>
    <div id="root"></div>
    <script>
      if (typeof process === 'object') {
        const scripts = [];

        if (process.env.NODE_ENV === 'development') {
          // Dynamically insert the DLL script in development env in the
          // renderer process
          scripts.push('../dll/renderer.dev.dll.js');
        }
        if (process.env.START_HOT) {
          // Dynamically insert the bundled app script in the renderer process
          const port = process.env.PORT || 3001;
          scripts.push(`http://localhost:${port}/dist/renderer.dev.js`);
        } else {
          scripts.push('../dist/renderer.prod.js?v' + Math.random());
        }

        if (scripts.length) {
          document.write(
            scripts
              .map(script => `<script defer src="${script}"><\/script>`)
              .join('')
          );
        }
      }
    </script>
  </body>
</html>
