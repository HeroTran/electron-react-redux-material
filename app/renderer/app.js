import * as React from 'react';
import { HashRouter, Switch, Route } from 'react-router-dom';
import PrivateRoute from './components/Common/PrivateRoute/PrivateRoute';
import { LoginContainer, AppContainer } from './containers';
import { ROUTE_PAGE } from './functionals/Common/constants';

const App = () => {
  return (
    <HashRouter>
      <Switch>
        <Route path={ROUTE_PAGE.ROUTE_LOGIN} component={LoginContainer} />
        <PrivateRoute path={ROUTE_PAGE.ROUTE_HOME} component={AppContainer} />
      </Switch>
    </HashRouter>
  );
};
export default App;
