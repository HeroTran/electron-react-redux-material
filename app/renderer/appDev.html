<!DOCTYPE html>
<html lang="en">
  <head>
    <title>herotran210_desktop_app</title>
    <meta charset="UTF-8">
    <meta name="description" content="herotran210_desktop_app">
    <meta name="keywords" content="herotran210_desktop_app">
    <meta name="author" content="herotran210_desktop_app">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <link rel="icon" type="image/png" sizes="32x32" href="../../resources/favicon.ico" />
    <link rel="icon" type="image/png" sizes="16x16" href="../../resources/favicon.ico" />
    <link rel="shortcut icon"  href="../../resources/favicon.ico" />
  </head>
  <body>
    <div id="root"></div>
    <script>
      if (typeof process === 'object') {
        const scripts = [];
        const port = process.env.PORT || 3001;
        scripts.push('../dll/renderer.dev.dll.js?v' + Math.random());
        scripts.push(
          `http://localhost:${port}/dist/renderer.dev.js?v` + Math.random()
        );

        if (scripts.length) {
          document.write(
            scripts
              .map((script) => `<script defer src="${script}"><\/script>`)
              .join('')
          );
        }
      }
    </script>
  </body>
</html>
