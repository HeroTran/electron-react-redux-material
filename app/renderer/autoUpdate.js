const { ipcRenderer } = require('electron');

const select = (selector) => document.querySelector(selector);

const container = select('#messages');
const progressBar = select('#progressBar');
const version = select('#version');

ipcRenderer.on('message', (event, text) => {
  const message = document.createElement('div');
  message.innerHTML = text;
  container.appendChild(message);
});

ipcRenderer.send('app_version');
ipcRenderer.on('app_version', (event, arg) => {
  ipcRenderer.removeAllListeners('app_version');
  version.innerText = `Version ${arg.version}`;
});

ipcRenderer.on('download-progress', (event, text) => {
  progressBar.style.width = `${text}%`;
});

ipcRenderer.on('update-downloaded', () => {
  ipcRenderer.send('restart_app');
});
