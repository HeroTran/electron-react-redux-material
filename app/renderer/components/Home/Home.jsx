import * as React from 'react';
import './home.scss';

const Home = (props) => {
  return (
    <div className="main-home">
      <h1>Home</h1>
    </div>
  );
};

export default Home;
