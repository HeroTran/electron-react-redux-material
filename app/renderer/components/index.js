import App from './App/App';
import Login from './AuthUser/Login';
import ClassList from './ClassList/ClassList';
import Home from './Home/Home';
import NoMatch from './Common/NoMatch/NoMatch';
import PrivateRoute from './Common/PrivateRoute/PrivateRoute';
import OpenSlot from './OpenSlot/OpenSlot';
import Profile from './Profile/Profile';
import Setting from './Setting/Setting';
import VirtualClassRoom from './VirtualClassRoom/VirtualClassRoom';

export {
  App,
  Login,
  ClassList,
  NoMatch,
  PrivateRoute,
  Home,
  OpenSlot,
  Profile,
  Setting,
  VirtualClassRoom,
};
