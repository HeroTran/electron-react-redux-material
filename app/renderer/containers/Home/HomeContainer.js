import { connect } from 'react-redux';
import { withTranslation } from 'react-i18next';
import Home from '../../components/Home/Home';

const mapStateToProps = (state) => ({});

const mapDispatchToProps = (dispatch) => ({});

const HomeInil = withTranslation('translation')(Home);
export default connect(mapStateToProps, mapDispatchToProps)(HomeInil);
