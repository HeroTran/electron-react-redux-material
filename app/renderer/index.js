import * as React from 'react';
import * as ReactDOM from 'react-dom';
import { Provider } from 'react-redux';
import { CssBaseline } from '@material-ui/core';
import { createMuiTheme, MuiThemeProvider } from '@material-ui/core/styles';
import { ConnectedRouter } from 'connected-react-router/immutable';
import { I18nextProvider } from 'react-i18next';
import configureStore, { history } from './functionals/store';
import App from './app';
import i18n from './i18n';
import './styles/styles.scss';

export const store = configureStore();
const theme = createMuiTheme({
  typography: {
    useNextVariants: true,
  },
});
const render = () => {
  ReactDOM.render(
    <Provider store={store}>
      <MuiThemeProvider theme={theme}>
        <CssBaseline />
        <ConnectedRouter history={history}>
          <I18nextProvider i18n={i18n}>
            <App />
          </I18nextProvider>
        </ConnectedRouter>
      </MuiThemeProvider>
    </Provider>,
    document.getElementById('root') || document.createElement('div') // for testing purposes
  );
};
render();
if (module.hot) {
  module.hot.accept(['./i18n', './components/App/App'], () => {
    ReactDOM.unmountComponentAtNode(document.getElementById('root'));
    render();
  });
}
