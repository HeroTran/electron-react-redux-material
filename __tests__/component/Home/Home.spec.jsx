import React from 'react';
import renderer from 'react-test-renderer';
import Home from '../../../app/renderer/components/Home/Home';

describe('should be rendered without error', () => {
  it('matches snapshot', () => {
    const tree = renderer.create(<Home />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
