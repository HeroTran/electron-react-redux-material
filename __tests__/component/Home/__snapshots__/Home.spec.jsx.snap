// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should be rendered without error matches snapshot 1`] = `
<div
  className="main-home"
>
  <h1>
    Home
  </h1>
</div>
`;
